plugins/module_utils/_vmware_facts.py

```diff
@@ -281,6 +281,128 @@ def hw_network_device_facts(self):
+        }
+
+    def identifier_facts(self):
+        return {
+            "moid": self.cluster._moId,
+            "datacenter": self.cluster.parent.parent.name
+        }
+
+    def host_facts(self):
+        hosts = []
+        for host in self.cluster.host:
+            hosts.append({
+                'name': host.name,
+                'folder': get_folder_path_of_vm(host),
+            })
+        return {"hosts": hosts}
+
+    def ha_facts(self):
+        fact_keys = [
+            "ha_enabled", "ha_failover_level", "ha_vm_monitoring", "ha_admission_control_enabled",
+            "ha_restart_priority", "ha_vm_tools_monitoring", "ha_vm_min_up_time", "ha_vm_max_failures",
+            "ha_vm_max_failure_window", "ha_vm_failure_interval"
+        ]
+        ha_facts = dict.fromkeys(fact_keys, None)
+        das_config = self.cluster.configurationEx.dasConfig
+        if not das_config:
+            ha_facts['ha_enabled'] = False
+            return ha_facts
+
+        ha_facts["ha_enabled"] = das_config.enabled
+        ha_facts["ha_vm_monitoring"] = das_config.vmMonitoring
+        ha_facts["ha_host_monitoring"] = das_config.hostMonitoring
+        ha_facts["ha_admission_control_enabled"] = das_config.admissionControlEnabled
+
+        if getattr(das_config, "admissionControlPolicy"):
+            ha_facts['ha_failover_level'] = das_config.admissionControlPolicy.failoverLevel
+        if getattr(das_config, "defaultVmSettings"):
+            ha_facts['ha_restart_priority'] = das_config.defaultVmSettings.restartPriority
+            ha_facts['ha_vm_tools_monitoring'] = das_config.defaultVmSettings.vmToolsMonitoringSettings.vmMonitoring
+            ha_facts['ha_vm_min_up_time'] = das_config.defaultVmSettings.vmToolsMonitoringSettings.minUpTime
+            ha_facts['ha_vm_max_failures'] = das_config.defaultVmSettings.vmToolsMonitoringSettings.maxFailures
+            ha_facts['ha_vm_max_failure_window'] = das_config.defaultVmSettings.vmToolsMonitoringSettings.maxFailureWindow
+            ha_facts['ha_vm_failure_interval'] = das_config.defaultVmSettings.vmToolsMonitoringSettings.failureInterval
+
+        return ha_facts
+
+    def dpm_facts(self):
+        fact_keys = ["dpm_enabled", "dpm_default_dpm_behavior", "dpm_host_power_action_rate"]
+        output_facts = dict.fromkeys(fact_keys, None)
+        dpm_config = self.cluster.configurationEx.dpmConfigInfo
+        if not dpm_config:
+            output_facts["dpm_enabled"] = False
+            return output_facts
+
+        output_facts["dpm_enabled"] = dpm_config.enabled
+        output_facts["dpm_default_dpm_behavior"] = getattr(dpm_config, "defaultDpmBehavior", None)
+        # dpm host power rate is reversed by the vsphere API. so a 1 in the API is really a 5 in the UI
+        try:
+            output_facts["dpm_host_power_action_rate"] = 6 - int(dpm_config.hostPowerActionRate)
+        except (TypeError, AttributeError):
+            output_facts["dpm_host_power_action_rate"] = 3
```

Comment:
Can you use consts here? (for the values 3 as a default and for 6)

Reply:
I made a constant for the default value, and a static method for the adjustment (6 - whatever).
I updated the drs and dpm modules to use those instead of having their own copies